### DISCUSSION TITLE
Is it possibile to simulate CCCV current_interpolant from an input file using pybamm?

### DISCUSSION THREAD

**Discussion by ssingh-ipa** [pybamm-team/PyBaMM#3509] [Q&A] [answered]
![Capture](https://github.com/pybamm-team/PyBaMM/assets/65173961/42dfc94e-d0ab-4b05-81be-604d76cc04cf)

I have given an interpolated input current from an excel file which has CC discharge and CCCV charge curve. Once the simulation reaches the "Lower voltage cut-off [V]": 2.8, it stops there and doesn't move ahead. The next section of applied current is the rest period of current = 0A and then the CC current is again applied. Is pybamm not capable of recognizing rest CC phase, rest phase and CV phase?

**Comment by TomTranter** [comment, marked as answer]
I would suggest splitting the input currents into 4 blocks and having 4 steps and then if a voltage limit is triggered it should move to the next step

**Reply by ssingh-ipa** [reply]
That's a good idea. I tested it out and it works now. Thank you. 

However taking the inputs from Excel, as they are, would have been ideal. Do you know why we cannot give such a continuous input to the interpolated current function?

**Reply by TomTranter** [reply]
Well the model fails too far below a voltage limit as things become infeasible. It looked like you hadn't sized the cell correctly so got to a lower voltage quicker than it should have done. If you were to keep supplying a discharge current after the cut-off that wouldn't work so what should be the expected behaviour? To ignore current inputs when voltage conditions are met? There would be no way to tell the simulation to move to the next step with a continuous current

**Reply by ssingh-ipa** [reply]
By sizing the cell correctly, do you mean the geometrical parameters? I have updated them as follows:

        "Cell cooling surface area [m2]": 0.0040192,
        "Cell volume [m3]": 3.676e-04,
        "Electrode height [m]": 0.082,
        "Electrode width [m]": 5.7,
        "Negative current collector thickness [m]": 10e-6,
        "Negative electrode thickness [m]": 67.3e-6,
        "Negative particle radius [m]": 6.72e-6,
        "Positive current collector thickness [m]": 2.0e-6,
        "Positive electrode thickness [m]": 55.5e-6,
        "Positive particle radius [m]": 3.28e-6,
        "Separator thickness [m]": 15e-6,
        'Number of electrodes connected in parallel to make a cell': 2.0,  

Is there anything further I need to consider for the cell sizing? I thought I needed to run further optimization algorithms on the entire parameter list because my voltage response in inaccurate. Do you think this incorrect response is mainly due to wrong sizing? What else can I try?

**Reply by TomTranter** [reply]
Just looking at the voltage plots the blue data seems to have about 50% extra capacity assuming the currents are identical. As well as thickness you should look at active material volume fraction and initial stoichiometries. If you start the simulation with a sub-optimal amount of lithium in each electrode then you can't reach full capacity.
